src/libraries/Directory.Build.props

```diff
@@ -215,12 +215,14 @@
     
   <!-- Set up Default symbol and optimization for Configuration -->
   <Choose>
-    <When Condition="'$(ConfigurationGroup)'=='Debug'">
+    <When Condition="'$(ConfigurationGroup)' == 'Debug'">
       <PropertyGroup>
         <DebugSymbols Condition="'$(DebugSymbols)' == ''">true</DebugSymbols>
         <Optimize Condition="'$(Optimize)' == ''">false</Optimize>
         <DebugType Condition="'$(DebugType)' == ''">full</DebugType>
-        <DefineConstants>$(DefineConstants),DEBUG,TRACE</DefineConstants>
+        <!-- Test projects should be configuration agnostic. -->
+        <DefineConstants Condition="'$(IsTestProject)' != 'true'">$(DefineConstants),DEBUG</DefineConstants>
+        <DefineConstants>$(DefineConstants),TRACE</DefineConstants>
```

Comment:
Is there a good reason to keep `TRACE` defined?

Reply:
It doesn't seem like it. From doing a grep, I saw that there are only a few usages of this constant, but the projects that use it, either define it on the files or on the csproj itself.

https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceSource.cs#L175

Should we just remove those conditionals and defines? It seems like it is always defined, therefore, those methods are always available?